servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -405,6 +405,47 @@ public boolean insertDomainTemplate(String domainName, String templateName, Stri
         return true;
     }
 
+    @Override
+    public boolean updateDomainTemplate(String domainName, String templateName, TemplateMetaData templateMetaData) {
+        DomainStruct domainStruct = getDomainStruct(domainName);
+        ArrayList<TemplateMetaData> templateMetalist = new ArrayList<>();
+        if (domainStruct == null) {
+            throw ZMSUtils.error(ResourceException.NOT_FOUND, "domain not found", "updateDomainTemplate");
+        }
+        if (domainStruct.getTemplates() == null) {
+            domainStruct.setTemplates(new ArrayList<>());
+        }
+        ArrayList<String> templates = domainStruct.getTemplates();
+        if (!templates.contains(templateName)) {
+            templates.add(templateName);
+        }
+
+        if (domainStruct.getTemplateMeta() == null) {
+            domainStruct.setTemplateMeta(new ArrayList<>());
+        }
+        ArrayList<TemplateMetaData> templateMetaList = domainStruct.getTemplateMeta();
+        if (!templateMetaList.isEmpty()) {
+            for (TemplateMetaData meta : templateMetaList) {
```

Comment:
we're missing the handling of the when when templateMetaList is not empty but it doesn't contain the templateName we're updating. In that case, we also need to add a new object.

Reply:
Done.